subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -38,14 +59,21 @@
 public interface TransformAction<T extends TransformParameters> {
 
     /**
-     * The object provided by {@link TransformSpec#getParameters()}.
+     * The object provided by {@link TransformSpec#getParameters()} when registering the artifact transform.
+     *
+     * <p>
+     *     Do not implement this method in your subclass.
+     *     Gradle provides the implementation when registering the transform action via {@link org.gradle.api.artifacts.dsl.DependencyHandler#registerTransform(Class, Action)}.
+     * </p>
      */
     @Inject
     T getParameters();
 
     /**
      * Executes the transform.
      *
+     * <p>This method must be overwritten in the subclass.</p>
```

Comment:
```suggestion
     * <p>This method must be implemented in the subclass.</p>
```

Do we need to explain this, though?

Reply:
I wanted to point it out in contrast to the `getParameters` method.

Reply:
👍